lib/retrieval-stats.js

```diff
@@ -49,9 +55,13 @@ export const buildRetrievalStats = (measurements, telemetryPoint) => {
     // `retrievalResult` should be always set by lib/preprocess.js, so we should never encounter
     // `UNDEFINED` result. However, I am still handling that edge case for extra robustness.
     const result = m.retrievalResult ?? 'UNDEFINED'
-    const oldCount = resultBreakdown[result] ?? 0
+    let oldCount = resultBreakdown[result] ?? 0
     resultBreakdown[result] = oldCount + 1
 
+    const indexerResult = m.indexerResult ?? 'UNDEFINED'
+    oldCount = indexerResultBreakdown[indexerResult] ?? 0
+    indexerResultBreakdown[indexerResult] = oldCount + 1
```

Comment:
two alternatives to consider, which don't repurpose the `oldCount` variable (risky imo):

```suggestion
    resultBreakdown[result] = (resultBreakdown[result] ?? 0) + 1

    const indexerResult = m.indexerResult ?? 'UNDEFINED'
    indexerResultBreakdown[indexerResult] = (indexerResultBreakdown[indexerResult] ?? 0) + 1
```

```suggestion
    {
      const oldCount = resultBreakdown[result] ?? 0
      resultBreakdown[result] = oldCount + 1
    }

    {
      const indexerResult = m.indexerResult ?? 'UNDEFINED'
      const oldCount = indexerResultBreakdown[indexerResult] ?? 0
      indexerResultBreakdown[indexerResult] = oldCount + 1
    }
```

Reply:
Yeah, I making `oldCount` mutable is a code smell.

I decided to extract a helper function - see 6b88ceff